src/auxiliary/Filesystem.cpp

```diff
@@ -168,5 +171,92 @@ remove_file( std::string const& path )
 #endif
 }
 
-} // auxiliary
-} // openPMD
+#if openPMD_HAVE_MPI
+
+namespace
+{
+    template< typename >
+    struct MPI_Types;
+
+    template<>
+    struct MPI_Types< unsigned long >
+    {
+        static MPI_Datatype const value;
+    };
+
+    template<>
+    struct MPI_Types< unsigned long long >
+    {
+        static MPI_Datatype const value;
+    };
+
+    template<>
+    struct MPI_Types< unsigned >
+    {
+        static MPI_Datatype const value;
+    };
+
+    /*
+     * Only some of these are actually instanciated,
+     * so suppress warnings for the others.
+     */
+    __attribute__((unused))
```

Comment:
This is GNU-ish known but a compiler-extension to std C++.
Please guard as usual via helpers like this:
https://github.com/openPMD/openPMD-api/blob/0.12.0-alpha/include/openPMD/auxiliary/Deprecated.hpp

Ref.: https://stackoverflow.com/a/31909713/2719194